packages/astro/src/assets/vendor/image-size/types/pnm.ts

```diff
@@ -19,7 +19,7 @@ const handlers: { [type: string]: Handler} = {
     let dimensions: string[] = []
 
     while (lines.length > 0) {
-      const line = lines.shift() as string
+      const line = lines.shift()!
```

Comment:
The bang operator is unsafe; I have a practice of leaving a `// SAFETY: <explanation>` comment where I explain why it's safe to do these kinds of casting.

Reply:
Yeah, both approaches require attention. I believe this is a chore from our side

Reply:
Yup - `!` is precisely the same as `as string` when used on a nullish `string`. All this change is doing is going with a slightly more succinct syntax. https://typescript-eslint.io/rules/non-nullable-type-assertion-style

Reply:
Thank you. I suppose we can keep the existing changes, then we can enable the lint rule and then we can refactor the errors :) 

The bang operator is easily detectable by an analyzer

Reply:
Hi, I wrote the original PR to Vite replacing `as` with `!`. The reason for this is that while both can make potentially unsafe assertions, `as` can silence type errors when types are assignable, making `!` relatively more safe.